### PR TITLE
Add vignette documenting hdf5 support 

### PR DESCRIPTION
**Pull request by kevinrue**

### CODE REVIEW

**Comment by kevinrue** [comment]
@LTLA if you just want to polish the terminology of HDF5 and DelayedArray, then we'll be good to merge

File: vignettes/hdf5.Rmd [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+sce <- readRDS("sce.rds")
+```
+
+```{r, eval=!exists("SCREENSHOT"), include=FALSE}
+SCREENSHOT <- function(x, ...) knitr::include_graphics(x)
+```
+
+# Wrapping matrices in HDF5 files
+
+Let us make a copy of the originak `sce` object to work with; we will be comparing the memory footprint of the object using an HDF5 backend.
```
Comment (LTLA) [inline vignettes/hdf5.Rmd:66]: original

File: vignettes/hdf5.Rmd [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+```{r}
+sce_h5 <- sce
+```
+
+Many `SummarizedExperiment` objects store assay matrices as regular `matrix` objects.
```
Comment (LTLA) [inline vignettes/hdf5.Rmd:72]: Or sparse matrices. Basically anything in memory.

File: vignettes/hdf5.Rmd [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+```{r}
+library(SingleCellExperiment)
+class(assay(sce_h5, "logcounts"))
```
Comment (LTLA) [inline vignettes/hdf5.Rmd:76]: Bit confusing to see `sce_h5` with an in-memory matrix. Suggest deferring the creation of a new SE variable until you actually create a HDF5Matrix backend.

File: vignettes/hdf5.Rmd [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+```{r}
+library(HDF5Array)
+assay(sce_h5, "tophat_counts") <- writeHDF5Array(assay(sce_h5, "tophat_counts"))
+class(assay(sce_h5, "tophat_counts"))
+assay(sce_h5, "logcounts") <- writeHDF5Array(assay(sce_h5, "logcounts"))
+class(assay(sce_h5, "logcounts"))
+```
+
+Comparing the memory footprint of the backed up original object `sce` and the HDF5-backed `sce_h5` object reveals the smallermemory footprint of HDF5-backed objects.
```
Comment (LTLA) [inline vignettes/hdf5.Rmd:96]: Typo. And it's not just smaller - it's effectively constant size no matter the size of the dataset, with everything just being pushed to the disk.

File: vignettes/hdf5.Rmd [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+```{r, results='asis', echo=FALSE}
+cat(sprintf(
+  "In this case, using an HDF5 backend reduced the memory footprint %.1f-fold.",
+  object.size(sce) / object.size(sce_h5)))
+```
+
+# iSEE supports HDF5-backed assays
+
+We can then used the `SingleCellExperiment` object as usual.
```
Comment (LTLA) [inline vignettes/hdf5.Rmd:111]: Mention that this works for any (i) DelayedArray backend (e.g., `ResidualMatrix`, `LowRankMatrix` from the **BiocSingular** package; `RleMatrix` objects; SQL-backed stores, though I don't know if these exist yet) and (ii) any matrix-like object in general that supports `dim()` and `[`.

**Comment by kevinrue** [comment]
Edited following review. Check again please.

**Comment by LTLA** [comment]
I reorganized the thrust of the vignette to be more focused on big data. Can you drag in the downsampling description and add a screenshot describing how to set it and to show how it doesn't have a deleterious visual effect?

**Pull request merged**